sematic/api/endpoints/runs.py

```diff
@@ -413,3 +414,21 @@ def save_graph_endpoint(user: Optional[User]):
     #    return jsonify_error(str(e), HTTPStatus.INTERNAL_SERVER_ERROR)
 
     return flask.jsonify({})
+
+
+@sematic_api.route("/api/v1/runs/<run_id>/external_resources", methods=["POST"])
+@authenticate
+def link_resource_endpoint(user: Optional[User], run_id: str) -> flask.Response:
+    if (
+        not flask.request
+        or not flask.request.json
+        or "external_resource_ids" not in flask.request.json
+    ):
+        return jsonify_error(
+            "Please provide an external_resource_id payload with the request",
+            HTTPStatus.BAD_REQUEST,
+        )
+    external_resource_ids = flask.request.json["external_resource_ids"]
+    for resource_id in external_resource_ids:
+        save_run_external_resource_link(resource_id=resource_id, run_id=run_id)
```

Comment:
Same here, can we reduce from N queries to 1?

Reply:
done